### PR TITLE
Remove dead links in login modal for skipUserManagement

### PR DESCRIPTION
**Pull request by cbornet**
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/shared/login/login.component.html.ejs [hunk 1]
```diff
@@ -40,7 +40,7 @@
                     <input type="password" class="form-control" name="password" id="password" placeholder="{{'login.form.password.placeholder' | translate}}"
                            formControlName="password">
                 </div>
-                <div class="form-check">
+                <div class="form-check"<% if (authenticationType === 'session' && databaseType === 'no') { %> hidden<% } %>>
```
Comment (wmarques) [inline generators/client/templates/angular/src/main/webapp/app/shared/login/login.component.html.ejs]: is this related to `skipUserManagement`?

Comment (cbornet) [reply generators/client/templates/angular/src/main/webapp/app/shared/login/login.component.html.ejs]: it's more related to `no database`.
